server/portal/apps/projects/views.py

```diff
@@ -0,0 +1,264 @@
+from portal.apps.projects.managers.base import ProjectsManager
+# from portal.apps.search.api.managers.project_search import ProjectSearchManager
+
+
+LOGGER = logging.getLogger(__name__)
+
+
+@python_2_unicode_compatible
+@method_decorator(agave_jwt_login, name='dispatch')
+@method_decorator(login_required, name='dispatch')
+class ProjectsApiView(BaseApiView):
+    """Projects API view.
+
+    This view handles anything that has to do with multiple projects.
+    Creating a project is implemented here since the call does not
+    specify an id. Creating a project id takes into consideration the rest
+    of the projects.
+    """
+
+    def get(self, request):
+        """GET handler.
+
+        If no 'query_string' is present this view will return a list of every
+        project where the requesting user is a part of.
+        If a `query_string` value is present (e.g.
+        ``GET /api/projects/?query_string="vertigo"```) then the list of
+        projects returned are the projects where the requesting user is a
+        part of AND the query string is present in any of its fields.
+
+        Sample response:
+        ```json
+        {"response": [{
+            "absolutePath": "/corral-repl/tacc/aci/CEP/projects/CEP-7",
+            "available": true,
+            "default": false,
+            "description": "Project Title",
+            "globalDefault": false,
+            "id": "cep.project.CEP-7",
+            "name": "CEP-7",
+            "owner": null,
+            "public": false,
+            "revision": null,
+            "site": null,
+            "status": "UP",
+            "storage": {
+                "auth": {
+                    "password": null,
+                    "privateKey": null,
+                    "publicKey": null,
+                    "type": null,
+                    "username": null
+                },
+                "homeDir": null,
+                "host": null,
+                "mirror": false,
+                "port": null,
+                "protocol": null,
+                "proxy": null,
+                "publicAppsDir": null,
+                "rootDir": null
+            },
+            "type": "STORAGE",
+            "uuid": null
+        }, ... ],
+        "status": 200
+        }
+        ```
+        """
+        query_string = request.GET.get('query_string')
+        offset = int(request.GET.get('offset', 0))
+        limit = int(request.GET.get('limit', 100))
+
+        mgr = ProjectsManager(request.user)
+
+        if query_string is not None:
+            # search_mgr = ProjectSearchManager(username=request.user.username, query_string=query_string)
+            # search_mgr.search(offset=offset, limit=limit)
+            # res = search_mgr.list(mgr=mgr)
+            return
+        else:
+            res = mgr.list(
+                offset=offset,
+                limit=limit
+            )
+        return JsonResponse(
+            {
+                'status': 200,
```

Comment:
In these responses from Cepv1 code, it is kinda weird that there is a `status` in the json object as the JsonReponse also has a `status` attribute which defaults (in this case) to 200.

I wonder if we should use this or a subsequent pr to refactor the JsonReponses (kinda related to https://jira.tacc.utexas.edu/browse/FP-111) to just:
```
        return JsonResponse(
            {
                'response': res,
            },
            encoder=ProjectsManager.meta_serializer_cls
        )
```